driver-core/src/test/unit/com/mongodb/internal/operation/QueryBatchCursorSpecification.groovy

```diff
@@ -179,24 +177,14 @@ class QueryBatchCursorSpecification extends Specification {
 
         then:
         // simulate the user calling `close` while `getMore` is in flight
-        if (useCommand) {
-            // in LB mode the same connection is used to execute both `getMore` and `killCursors`
-            int numberOfInvocations = serverType == ServerType.LOAD_BALANCER
-                    ? getMoreResponseHasCursor ? 2 : 1
-                    : 1
-            numberOfInvocations * conn.command(*_) >> {
-                // `getMore` command
-                cursor.close()
-                getMoreResponse
-            } >> {
-                // `killCursors` command
-                null
-            }
-        } else {
-            1 * conn.getMore(*_) >> {
-                cursor.close()
-                getMoreResponse
-            }
+        // in LB mode the same connection is used to execute both `getMore` and `killCursors`
+        numberOfInvocations * conn.command(*_) >> {
+            // `getMore` command
+            cursor.close()
+            getMoreResponse
+        } >> {
+            // `killCursors` command
+            null
         }
```

Comment:
The reason this test fails with the change made in the PR, is that it assumes that the same `conn` is used for issuing the `killCursors` regardless of whether LB mode is used (and the connection is pinned to the cursor) or not. I don't think this assumption is valid for a non-LB mode.

If the mocking in test done without the aforementioned assumption for non-LB mode, the test passes when run against `QueryBatchCursor` without requiring any more changes. See https://github.com/stIncMale/mongo-java-driver/commit/a6cad08e4fa147ca179e8dffc33e5871daf4b34d for the proposed change.